java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java

```diff
@@ -139,6 +139,7 @@ public synchronized PoiSearchResult searchPoi(SearchService.PoiSearchData data)
             }
             leftoverLimit = limit - res.size();
         }
+        osmAndMapsService.obfFiles.values().forEach(ref -> mapList.forEach(ref::unlockReader));
```

Comment:
**Use pattern** should be inside 1 method
try{ 
 lock
} finally {
 unlock
}